examples/platform/esp32/diagnostics/diagnostic-logs-provider-delegate-impl.cpp

```diff
@@ -31,7 +32,6 @@ LogProvider::CrashLogContext LogProvider::sCrashLogContext;
 
 #ifdef CONFIG_ENABLE_ESP_DIAGNOSTICS_TRACE
```

Comment:
Is this option required here? Only for the `sReadEntries` variable. Instead should these files be guarded entirely with this option?